src/pendulum_visualizer.py

```diff
@@ -39,23 +41,33 @@ def __init__(self):
         self.center_of_mass = self.ax.plot(0, -self.ac1, zorder=1, color='b', marker='o', markersize=14)
 
     def draw(self, context):
-        theta = self.EvalVectorInput(context, 0).get_value()[0]
+        if isinstance(context, Context):
```

Comment:
I think it would be cleaner to have the draw method always take theta / positions and have the system method that calls draw pull them from the context (for this and fhe planarrigidbodyviz)?

Reply:
It's the superclass that invokes the Draw method on every _DoPublish() -- but I've been intentionally keeping assumptions of the input type away from that superclass. Things (like animate) might simplify if we can find a way to hold that information in PyPlotVisualizer sufficiently generally to not be restrictive...

(Or every subclass can reimplment DoPublish, but then some of PyPlotVisualizer's internal figure handling might leak down, and the subclasses get more complicated.)

Reply:
Good point.  Ok